File: OOP_PYTHON/L1_OCIP/Student.py
class Student:

    def __init__(self, name, father_name, nationality, city, obtained_marks, total_marks):
        self.name = name
        self.father_name = father_name
        self.city = city
        self.nationality = nationality
        self.obtained_marks = obtained_marks
        self.total_marks = total_marks

    def personal_detail(self):
        print("INFORMATION FOR SHERAZ")
        print("NAME        : ", self.name)
        print("FATHER NAME : ", self.father_name)
        print("CITY        : ", self.city)
        print("NATIONALITY : ", self.nationality)
        print("MARKS       : ", self.obtained_marks)
        print("TOTAL       : ", self.total_marks)
        print("GRADE       : ", self.calculate_grade())
        print()

    def calculate_grade(self):
        if 50 <= self.obtained_marks < 75:
            return "C"
        elif 75 <= self.obtained_marks < 90:
            return "B"
        elif 90 <= self.obtained_marks < 100:
            return "A"
        return "F"

File: OOP_JAVA/L1_OCIP/src/objectandclass/Student.java
package objectandclass;

public class Student {

    // ATTRIBUTES/DATA-MEMBERS      -- STATE OF CLASS
    public String name;
    public String fatherName;
    public String city;
    public String nationality;
    public int obtainedMarks;
    public int totalMarks;

    // METHODS/FUNCTIONS/PROCEDURE  -- BEHAVIOUR OF CLASS

    public void registration(String name, String fatherName, String city, String nationality, int obtainedMarks){
        this.name=name;
        this.fatherName=fatherName;
        this.city=city;
        this.nationality=nationality;
        this.obtainedMarks=obtainedMarks;
    }

    public void showResults(){
        System.out.println("RESULTS FOR SHERAZ");
        System.out.println("MARKS      : "+ obtainedMarks);
        System.out.println("TOTAL      : "+ totalMarks);
        System.out.println("GRADE      : "+ calculateGrade());
        System.out.println();
    }

    public void personalDetails(){
        System.out.println("INFORMATION FOR SHERAZ");
        System.out.println("NAME        : "+ name);
        System.out.println("FATHER NAME : "+ fatherName);
        System.out.println("CITY        : "+ city);
        System.out.println("NATIONALITY : "+ nationality);
        System.out.println("MARKS       : "+ obtainedMarks);
        System.out.println("TOTAL       : "+ totalMarks);
        System.out.println("GRADE       : "+ calculateGrade());
        System.out.println();
    }

    public String calculateGrade(){
        if (obtainedMarks >= 50 && obtainedMarks < 75) {
            return "C";
        }else if (obtainedMarks >= 75 && obtainedMarks < 90) {
            return "B";
        }else if (obtainedMarks >= 90 && obtainedMarks < 100) {
            return "A";
        }
        return "F";
    }

}
